packages/react-reconciler/src/__tests__/ReactIncrementalErrorLogging-test.js

```diff
@@ -25,20 +31,28 @@ describe('ReactIncrementalErrorLogging', () => {
 
     const InternalTestUtils = require('internal-test-utils');
     waitForAll = InternalTestUtils.waitForAll;
-    waitForThrow = InternalTestUtils.waitForThrow;
   });
 
   // Note: in this test file we won't be using toErrorDev() matchers
   // because they filter out precisely the messages we want to test for.
+  let oldConsoleWarn;
   let oldConsoleError;
   beforeEach(() => {
+    oldConsoleWarn = console.warn;
     oldConsoleError = console.error;
+    console.warn = jest.fn();
     console.error = jest.fn();
+    uncaughtExceptionMock = jest.fn();
+    process.on('uncaughtException', uncaughtExceptionMock);
   });
 
   afterEach(() => {
+    console.warn = oldConsoleWarn;
     console.error = oldConsoleError;
+    process.off('uncaughtException', uncaughtExceptionMock);
```

Comment:
Should we use a jsdom environment here so we're testing the browser error handling?

Reply:
It's either way I guess. We test jsdom a lot elsewhere so maybe good to have some node coverage?